blob/service.go

```diff
@@ -86,6 +89,47 @@ func DefaultSubmitOptions() *SubmitOptions {
 	}
 }
 
+type BlobsubResponse struct {
+	blobs  []*Blob
+	height uint64
+}
+
+func (s *Service) Subscribe(ctx context.Context, ns share.Namespace) (<-chan *BlobsubResponse, error) {
+	headerCh, err := s.headerSub(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	blobCh := make(chan *BlobsubResponse)
+	go func() {
+		defer close(blobCh)
+
+		for {
+			select {
+			case header, ok := <-headerCh:
+				if !ok {
+					log.Errorw("header channel closed for blobsub", "namespace", ns.ID())
+					return
+				}
+				blobs, err := s.GetAll(ctx, header.Height(), []share.Namespace{ns})
+				if err != nil {
+					log.Errorw("failed to get blobs", "height", header.Height(), "namespace", ns.ID(), "err", err)
+					continue
+				}
+
+				select {
+				case <-ctx.Done():
+					return
```

Comment:
IMO you need also to respect the service context(which has not been used so far but we should add it with Start/Stop methods)